nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RPathUtils.java

```diff
@@ -115,6 +117,13 @@ else if (Pattern.compile(PATTERN_METADATA_RDS).matcher(path).matches()) {
     return assetKind;
   }
 
+  /**
+   * Determines if it's a valid path for archive
+   */
+  public static boolean isValidArchivePath(final String path) {
+    return Pattern.compile(PATTERN_ARCHIVE).matcher(path).matches();
```

Comment:
Minor: You could actually do the `Pattern.compile(PATTERN_ARCHIVE)` as a field member, so that it doesn't have to run `compile` each time, I believe.

Reply:
Nice suggestion, done in d659b9bb95d8f2914780f13608efcb4d1f2d83b9